kie-dmn-feel/src/main/java/org/kie/dmn/feel/parser/feel11/ParserHelper.java

```diff
@@ -69,18 +80,28 @@ public void recoverScope() {
     }
 
     public void recoverScope( String name ) {
+        LOG.trace("[{}] recoverScope( name: {}) with currentScope: {}", this.currentScope.getName(), name, currentScope);
         Scope s = this.currentScope.getChildScopes().get( name );
         if( s != null ) {
             currentScope = s;
         } else {
             pushScope();
```

Comment:
This I have some doubt, even with reference to original code.
Say it's compiling this FEEL expression:
```
{ myFeelVar: person.home address.street name
```

even in the original code, it would end up pushing as Contexts the following stack, topmost to bottom of stack:
```
[myFeelVar]
[myFeelVar]
[<local>]
[<global>]
[<built-in>]
```

hence I'm not 100% sure the relationship between the currentName stack and the currentContext stack?
considering when the method `pushStack` is invoked, creates indeed a new Scope using the current stack's peek name.